packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -68,30 +57,49 @@ const processFile = (file, jobs, cb) => {
       clonedPipeline = clonedPipeline.rotate(args.rotate)
     }
 
-    if (job.file.extension.match(/^jp/)) {
-      clonedPipeline.toFile(job.outputPath, (err, info) => {
-        bar.tick()
-        job.outsideResolve(info)
-      })
-      // Compress pngs
-    } else if (job.file.extension === `png`) {
-      clonedPipeline.toBuffer().then(sharpBuffer => {
-        imagemin
-          .buffer(sharpBuffer, {
-            plugins: [
-              imageminPngquant({
-                quality: `${args.quality}-${Math.min(args.quality + 25, 100)}`, // e.g. 40-65
-              }),
-            ],
-          })
-          .then(imageminBuffer => {
-            fs.writeFile(job.outputPath, imageminBuffer, () => {
-              bar.tick()
-              job.outsideResolve()
+    // duotone and write to file
+    duotone(
```

Comment:
It'd be clearer to check if duotone is set as an option here vs. inside duotone.js